### PR TITLE
PWN-5330 - Fix restore errors

### PR DESCRIPTION
**Pull request by davran1001**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-5330

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -157,6 +157,7 @@ android {
             signingConfig signingConfigs.release
             minifyEnabled true
             shrinkResources true
+            debuggable true
```
Comment (gslevinkov) [inline app/build.gradle:160]: revert

File: app/src/main/java/org/p2p/wallet/auth/interactor/restore/UserRestoreInteractor.kt [hunk 2]
```diff
@@ -169,31 +170,63 @@ class UserRestoreInteractor(
             }
         }
     } catch (e: Throwable) {
-        RestoreUserResult.RestoreFailure.DevicePlusSocialShare(RestoreUserException(e.message.orEmpty()))
+        val errorMessage = e.message.orEmpty()
+        Timber.e(errorMessage)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/interactor/restore/UserRestoreInteractor.kt:174]: won't log, use RestoreUserException

File: app/src/main/java/org/p2p/wallet/auth/interactor/restore/UserRestoreInteractor.kt [hunk 3]
```diff
@@ -169,31 +170,63 @@ class UserRestoreInteractor(
...
         if (result is RestoreUserResult.RestoreFailure) {
+            // if restore was failed
+            // Try last try with SOCIAL + CUSTOM
             result = tryRestoreUser(OnboardingFlow.RestoreWallet.SocialPlusCustomShare)
         }
-        return result
+
+        // if restore is FAILED we throw error thast SHARES ARE NOT MATCH
+        // otherwise Restore is SUCCESSFULL
+        return when (result) {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/interactor/restore/UserRestoreInteractor.kt:194]: if

File: app/src/main/java/org/p2p/wallet/auth/interactor/restore/UserRestoreInteractor.kt [hunk 4]
```diff
@@ -169,31 +170,63 @@ class UserRestoreInteractor(
...
+
         if (result is RestoreUserResult.RestoreFailure) {
+            // if restore was failed
+            // try last try with SOCIAL + CUSTOM
             result = tryRestoreUser(OnboardingFlow.RestoreWallet.SocialPlusCustomShare)
         }
-        return result
+        // if restore is FAILED we throw error thast SHARES ARE NOT MATCH
+        // otherwise Restore is SUCCESSFULL
+        return when (result) {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/interactor/restore/UserRestoreInteractor.kt:220]: if

File: app/src/main/java/org/p2p/wallet/auth/ui/generalerror/OnboardingGeneralErrorFragment.kt [hunk 5]
```diff
@@ -113,4 +150,41 @@ class OnboardingGeneralErrorFragment :
...
+    private fun handleSignResult(result: ActivityResult) {
+        signInHelper.parseSignInResult(requireContext(), result, errorHandler = this)?.let { credential ->
+            presenter.setGoogleIdToken(credential.id, credential.googleIdToken.orEmpty())
+        }
+    }
+
+    override fun onConnectionError() {
+        setRestoreByGoogleLoadingState(isLoading = false)
+        showUiKitSnackBar(message = getString(R.string.error_general_message))
+        TODO("Not yet implemented")
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/ui/generalerror/OnboardingGeneralErrorFragment.kt:179]: ???

File: app/src/main/res/values/onboarding_strings.xml [hunk 6]
```diff
@@ -87,10 +87,7 @@
     <string name="onboarding_general_error_starting_screen_button_title">Starting screen</string>
     <string name="onboarding_general_error_bug_report_button_title">Support</string>
     <string name="onboarding_general_error_critical_error_title">Well well 🦄</string>
-    <string name="onboarding_general_error_critical_error_sub_title">
-        We\'ve broke something really big. Let\'s wait together, finally the app will be repaired. \n\n
-        If you wish to report the issue, use error code #%1$s
-    </string>
+    <string name="onboarding_general_error_critical_error_sub_title">You’ve find a seldon page 🦄  It’s like a unicorn, but it’s a crush. We’re already fixing it.\n\nIf you will write us, use error code #%d</string>
```
Comment (gslevinkov) [inline app/src/main/res/values/onboarding_strings.xml:90]: old design texts

**Review by gslevinkov — CHANGES_REQUESTED** [review]

File: app/build.gradle [hunk 7]
```diff
@@ -157,6 +157,7 @@ android {
             signingConfig signingConfigs.release
             minifyEnabled true
             shrinkResources true
+            debuggable false
```
Comment (gslevinkov) [inline app/build.gradle:160]: remove at all

File: app/src/main/java/org/p2p/wallet/auth/interactor/restore/UserRestoreInteractor.kt [hunk 8]
```diff
@@ -50,12 +53,13 @@ class UserRestoreInteractor(
...
                 encryptedMnemonic = encryptedMnemonic
             )
+
             signUpDetailsStorage.save(
                 data = Web3AuthSignUpResponse(
                     ethereumPublicKey = result.ethereumPublicKey,
                     mnemonicPhrase = result.mnemonicPhrase,
                     encryptedMnemonicPhrase = JsonObject(),
-                    deviceShare = null,
+                    deviceShare = deviceShare,
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/interactor/restore/UserRestoreInteractor.kt:62]: why we are setting device share if we are doing SocialPlusCustomShare?

File: app/src/main/java/org/p2p/wallet/auth/ui/generalerror/OnboardingGeneralErrorFragment.kt [hunk 9]
```diff
@@ -118,4 +150,40 @@ class OnboardingGeneralErrorFragment :
...
+
+    private fun handleSignResult(result: ActivityResult) {
+        signInHelper.parseSignInResult(requireContext(), result, errorHandler = this)?.let { credential ->
+            presenter.setGoogleIdToken(credential.id, credential.googleIdToken.orEmpty())
+        }
+    }
+
+    override fun onConnectionError() {
+        setRestoreByGoogleLoadingState(isLoading = false)
+        showUiKitSnackBar(message = getString(R.string.error_general_message))
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/ui/generalerror/OnboardingGeneralErrorFragment.kt:178]: showUiKitSnackBar(messageResId = R.string.onboarding_offline_error)

**Review by gslevinkov — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/p2p/wallet/auth/interactor/restore/UserRestoreInteractor.kt (hunk 8 above)
Comment (davran1001) [reply app/src/main/java/org/p2p/wallet/auth/interactor/restore/UserRestoreInteractor.kt:62]: cause if user has device share on device, we should not clear it. Why ? I don't know)

File: app/src/main/res/values/onboarding_strings.xml [hunk 10]
```diff
@@ -86,11 +86,8 @@
     <string name="onboarding_general_error_terms_title">P2P App\'s</string>
     <string name="onboarding_general_error_starting_screen_button_title">Starting screen</string>
     <string name="onboarding_general_error_bug_report_button_title">Support</string>
-    <string name="onboarding_general_error_critical_error_title">Well well 🦄</string>
-    <string name="onboarding_general_error_critical_error_sub_title">
-        We\'ve broke something really big. Let\'s wait together, finally the app will be repaired. \n\n
-        If you wish to report the issue, use error code #%1$s
-    </string>
+    <string name="onboarding_general_error_critical_error_title">Well well</string>
+    <string name="onboarding_general_error_critical_error_sub_title">You’ve find a seldon page 🦄 It’s like a unicorn, but it’s a crush. We’re already fixing it.\n\nIf you will write us, use error code #%d</string>
```
[the remark this answers is not in the document]
Comment (davran1001) [reply app/src/main/res/values/onboarding_strings.xml:90]: QA said that it's right text

File: app/build.gradle (hunk 7 above)
Comment (gslevinkov) [inline app/build.gradle:160]:
```suggestion
```

**Review by gslevinkov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/auth/interactor/UserSignUpInteractor.kt [hunk 11]
```diff
@@ -56,7 +56,10 @@ class UserSignUpInteractor(
 
     private suspend fun generateDeviceAndThirdShare(): Web3AuthSignUpResponse {
         val torusKey = signUpFlowDataRepository.torusKey
-            ?: error("Torus key is null")
-        return web3AuthApi.triggerSilentSignUp(torusKey)
+        if (torusKey.isNullOrEmpty()) {
+            error("Torus key is null")
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/auth/interactor/UserSignUpInteractor.kt:60]: throw error("Torus key is null")?

**Review by kabramovp2p — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/auth/interactor/UserSignUpInteractor.kt (hunk 11 above)
Comment (davran1001) [reply app/src/main/java/org/p2p/wallet/auth/interactor/UserSignUpInteractor.kt:60]: public inline fun error(message: Any): Nothing = throw IllegalStateException(message.toString())

**Pull request merged**